internal/apiutil/transport.go

```
@@ -135,16 +135,28 @@ func ReadNumQuery[N number](r *http.Request, key string, def N) (N, error) {
 	switch any(def).(type) {
 	case int64:
 		v, err := strconv.ParseInt(val, 10, 64)
-		return N(v), err
+		if err != nil {
```

We should not mask actual errors here. We can either wrap them or inspect them and handle them in the error handler.